parts/plugins/apache.py

```diff
@@ -103,8 +103,8 @@ def __init__(self, name, options):
         super().__init__(name, options)
 
         self.build_packages.extend(['libapr1-dev', 'libaprutil1-dev',
-                                    'libpcre3-dev'])
-        self.stage_packages.extend(['libapr1', 'libaprutil1', 'libpcre3'])
+                                    'libpcre3-dev', 'libssl-dev', 'zlib1g-dev'])
```

Comment:
OK, so:

1. We can't rely on any package manager to give us the dependencies because we're compiling from source and have to figure it out from the manual compilation instructions
1. Libs are available in dev `.deb` packages
1. Those packages are added to the list of building dependencies
1. Each dev package has to be checked for sub-dependencies
1. Dev sub-dependencies don't need to be added

But then I don't get the rules. 
`libpcre3-dev` needs: `libc6-dev`, `libpcre3`, `libpcre16-3`, `libpcre32-3` , `libpcrecpp0v5`

Why aren't  `libpcre32-3` and `libpcrecpp0v5` added as staging dependencies?